certification/engine/engine_test.go

```diff
@@ -0,0 +1,56 @@
+package engine
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/runtime"
+)
+
+var _ = Describe("Engine Creation", func() {
+	Describe("When getting a new engine for a configuration", func() {
+		Context("with a valid configuration", func() {
+			cfg := runtime.Config{
+				Image:          "dummy/image",
+				EnabledChecks:  ContainerPolicy(),
+				ResponseFormat: "json",
+				LogFile:        "some.log",
+			}
+
+			It("should return and engine and no error", func() {
+				engine, err := NewForConfig(cfg)
+				Expect(err).To(BeNil())
+				Expect(engine).ToNot(BeNil())
```

Comment:
It might be good to compare the length of `CheckEngine.Checks` with the `EnabledCheck`.

Reply:
@samira-barouti Good catch. We actually can't do this reliably right now because we return a CheckRunner (interface) which does not expose a way of accessing the internal workings of the implementing type (I.e. we can't access `CheckEngine.Checks` from the result of `NewForConfig`.